fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNetworkMiddleware.kt

```diff
@@ -46,21 +50,82 @@ class ReviewQualityCheckNetworkMiddleware(
         store: Store<ReviewQualityCheckState, ReviewQualityCheckAction>,
         action: ReviewQualityCheckAction.NetworkAction,
     ) {
+        if (!networkChecker.isConnected()) {
+            store.updateProductReviewState(ProductReviewState.Error.NetworkError)
+            return
+        }
+
         when (action) {
             FetchProductAnalysis, RetryProductAnalysis -> {
                 scope.launch {
-                    val productReviewState = if (networkChecker.isConnected()) {
-                        reviewQualityCheckService.fetchProductReview().toProductReviewState()
-                    } else {
-                        ProductReviewState.Error.NetworkError
-                    }
-                    store.dispatch(ReviewQualityCheckAction.UpdateProductReview(productReviewState))
+                    logger.info("Fetching product analysis")
+                    val productReviewState = fetchAnalysis()
+                    logger.debug("Product review state: $productReviewState")
+                    store.updateProductReviewState(productReviewState)
                 }
             }
 
             ReviewQualityCheckAction.ReanalyzeProduct -> {
-                // Bug 1853311 - Integrate analyze and analysis_status
+                scope.launch {
```

Comment:
wondering aloud if the entire `when` should just be wrapped with `scope.launch{ }`

Reply:
that's a good idea!